ArcGISRuntimeSDKQt_CppSamples/Layers/BrowseOGCAPIFeatureService/BrowseOGCAPIFeatureService.cpp

```diff
@@ -0,0 +1,221 @@
+    m_collectionInfo.clear();
+    if (m_serviceInfo != nullptr)
+    {
+        delete m_serviceInfo;
+        m_serviceInfo = nullptr;
+    }
+    if (m_featureService != nullptr)
+    {
+        delete m_featureService;
+        m_featureService = nullptr;
+    }
+    m_featureCollectionList.clear();
+    emit featureCollectionListChanged();
+
+    // Instantiate new OGCFeatureService object using url
+    m_featureService = new OgcFeatureService(url, this);
+
+    // Connect loadStatusChanged and checkIfServiceLoaded()
+    connect(m_featureService, &OgcFeatureService::loadStatusChanged, this, &BrowseOGCAPIFeatureService::checkIfServiceLoaded);
+
+    // Connect errorOccurred to handleError
+    connect(m_featureService, &OgcFeatureService::errorOccurred, this, &BrowseOGCAPIFeatureService::handleError);
+
+    m_featureService->load();
+}
+
+void BrowseOGCAPIFeatureService::checkIfServiceLoaded(LoadStatus loadstatus)
+{
+    if (loadstatus != LoadStatus::Loaded)
+    {
+        return;
+    }
+    retrieveCollectionInfos();
+}
+
+void BrowseOGCAPIFeatureService::retrieveCollectionInfos()
+{
+    // Assign OGC service metadata to m_serviceInfo property
+    m_serviceInfo = m_featureService->serviceInfo();
+
+    // Assign information from each feature collection to the m_collectionInfo property
+    m_collectionInfo = m_serviceInfo->featureCollectionInfos();
+
+    createFeatureCollectionList();
+}
+
+void BrowseOGCAPIFeatureService::createFeatureCollectionList()
+{
+    m_featureCollectionList.clear();
+
+    // Populate list with names of each feature collection
+    for (OgcFeatureCollectionInfo* info : m_collectionInfo)
+    {
+        m_featureCollectionList.push_back(info->title());
+    }
+
+    emit featureCollectionListChanged();
+}
+
+MapQuickView* BrowseOGCAPIFeatureService::mapView() const
+{
+    return m_mapView;
+}
+
+// Set the view (created in QML)
+void BrowseOGCAPIFeatureService::setMapView(MapQuickView* mapView)
+{
+    if (!mapView || mapView == m_mapView)
+        return;
+
+    m_mapView = mapView;
+    m_mapView->setMap(m_map);
+
+    emit mapViewChanged();
+}
+
+QUrl BrowseOGCAPIFeatureService::featureServiceUrl() const
+{
+    return m_featureServiceUrl;
+}
+
+QStringList BrowseOGCAPIFeatureService::featureCollectionList() const
+{
+    return m_featureCollectionList;
+}
+
+QString BrowseOGCAPIFeatureService::errorMessage() const
+{
+    return m_errorMessage;
+}
+
+void BrowseOGCAPIFeatureService::loadService(QUrl urlFromInterface)
+{
+    m_featureServiceUrl = urlFromInterface;
+    emit urlChanged();
+    loadFeatureService(m_featureServiceUrl);
+}
+
+void BrowseOGCAPIFeatureService::loadFeatureCollection(int selectedFeature)
+{
+    // Create a CollectionInfo object from the selected collection
+    OgcFeatureCollectionInfo *info = m_collectionInfo[selectedFeature];
+
+    // Assign the CollectedInfo to the m_featureCollectionTable property
+    m_featureCollectionTable = new OgcFeatureCollectionTable(info);
```

Comment:
don't forget to set the parent QObject so that this object will get cleared up when it's parent is deleted
```suggestion
    m_featureCollectionTable = new OgcFeatureCollectionTable(info, this);
```

Reply:
I also wonder if you need to check if there already is a table and delete it before you assign?